src/Runtime/OMTensorList.inc

```diff
@@ -68,32 +64,29 @@ struct OMTensorList {
 
   int64_t _size; // Number of elements in _omts.
 
-  int64_t _owning; // indicates whether the OMTensorList owns the pointer to
-                   // OMTensor array or not. OMTensorList struct will release
-                   // the memory space referred to by '_omts' upon destruction
-                   // if and only if it owns it.
+  bool _shallow; // indicates whether we want to perform a shallow destroy or
+                 // regular destroy when calling the explicit
+                 // omTensorListDestroyShallow or
+                 //  omTensorListDestroy method.
 };
 
 /* OMTensorList creator */
 OMTensorList *omTensorListCreate(OMTensor **tensors, int64_t n) {
   OMTensorList *list = (OMTensorList *)malloc(sizeof(struct OMTensorList));
   if (!list)
     return NULL;
-  list->_omts = tensors;
   list->_size = n;
-  list->_owning = false;
-  return list;
-}
+  list->_shallow = false;
+  list->_omts = (OMTensor **)malloc(sizeof(OMTensor *) * n);
 
-/* OMTensorList creator with ownership */
-OMTensorList *omTensorListCreateWithOwnership(
-    OMTensor **tensors, int64_t n, int64_t owning) {
-  OMTensorList *list = (OMTensorList *)malloc(sizeof(struct OMTensorList));
-  if (!list)
+  if (!list->_omts) {
+    free(list); // free the previously allocated memory in case of an error
     return NULL;
-  list->_omts = tensors;
-  list->_size = n;
-  list->_owning = owning;
+  }
+  for (int64_t i = 0; i < list->_size; i++)
+    // copy the pointers to the OMTensor array
+    // now OMTensorList will always own the tensor array of pointers
+    list->_omts[i] = tensors[i];
```

Comment:
Do you know why `memcpy` doesn't work? It's simpler and more efficient code and that's what we want to use. If it doesn't, we need to find out why.

Reply:
See my previous comment… already stated my intent here

Reply:
@gongsu832 This comment is not called for and is actually very disrespectful.

‘That's what the code review is for. You seem to like trying different things without understanding why things work or don't work.’


This entire statement is not appropriate and unprofessional. It would behoove you to have facts before making statements. I asked you to talk and which you continue to ignore my request. So please if you want to be helpful and give advice do so in an appropriate manner because the commentary is not called for. You will actually DISCOURAGE people from contributing to community with such an attitude.

Reply:
@hamptonm1 My statements are based on what I observed. You certainly don't have to agree and continue the way you do things.

As far as "ignoring" you, everything I have to say is posted on the github so I don't know what else to talk about. I don't encourage or discourage people from contributing to community, that's their choice. I just don't want crappy code to go into the project.